### ISSUE TITLE
💅 lint/style/noParameterAssign should cover all or more cases

### ISSUE THREAD

**Issue by BC-M** [biomejs/biome#1577]
### Environment information

```bash
biome 1.5.2
```

### Rule name

lint/style/noParameterAssign

### Playground link

https://biomejs.dev/playground/?code=LwAvACAAdABoAGkAcwAgAHQAaAByAG8AdwBzACAAbgBvAFAAYQByAGEAbQBlAHQAZQByAEEAcwBzAGkAZwBuAAoAZgB1AG4AYwB0AGkAbwBuACAAYgBhAHIAKABmAG8AbwApACAAewAKACAAIABmAG8AbwAgAD0AIAAnAGYAbwBvACcAOwAKAH0ACgAKAC8ALwAgAHQAaABlAHMAZQAgAHMAaABvAHUAbABkACAAYQBsAHMAbwAgAHQAaAByAG8AdwAgAG4AbwBQAGEAcgBhAG0AZQB0AGUAcgBBAHMAcwBpAGcAbgAgAAoAZgB1AG4AYwB0AGkAbwBuACAAdABlAHMAdAAxACgAYgBhAHIAKQAgAHsACgAgACAAYgBhAHIALgBwAHIAbwBwACAAPQAgACIAdgBhAGwAdQBlACIAOwAKAH0ACgBmAHUAbgBjAHQAaQBvAG4AIAB0AGUAcwB0ADIAKABiAGEAcgApACAAewAKACAAIABkAGUAbABlAHQAZQAgAGIAYQByAC4AYQBhAGEAOwAKAH0ACgBmAHUAbgBjAHQAaQBvAG4AIAB0AGUAcwB0ADMAKABiAGEAcgApACAAewAKACAAIABiAGEAcgAuAGEAYQBhACsAKwA7AAoAfQAKAGYAdQBuAGMAdABpAG8AbgAgAHQAZQBzAHQANAAoAGIAYQByACkAIAB7AAoAIAAgAGIAYQByAFsAMABdACAAPQAgACIAdgBhAGwAdQBlACIAOwAKAH0ACgBmAHUAbgBjAHQAaQBvAG4AIAB0AGUAcwB0ADUAKABiAGEAcgApACAAewAKACAAIABiAGEAcgAuAHAAdQBzAGgAKAAiAHYAYQBsAHUAZQAiACkAOwAKACAAIABiAGEAcgAuAHAAbwBwACgAIgB2AGEAbAB1AGUAIgApADsACgB9AA%3D%3D

### Expected result

Should lint same cases like [eslint no-param-reassign](https://eslint.org/docs/latest/rules/no-param-reassign).

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
None of the “false negatives” you listed are reassignments though; they’re mutations of the parameter. While I agree a lint rule for that might be useful, I don’t think it should be covered by `noParameterAssign`.

Maybe a `noParameterMutation` rule would make sense? It’s rather difficult to enforce though, since we wouldn’t know which methods are mutating and which aren’t…

**Comment by Conaclos** [comment]
This is not a bug, this is an intentional behavior of the rule. I will convert the issue into a discussion.

As suggested by @arendjr, I think it is out of the scope of the rule.
